components/Search/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { ActionIcon, TextInput, useMantineColorScheme, useMantineTheme } from '@mantine/core';
+import { IconArrowLeft, IconArrowRight, IconSearch } from '@tabler/icons';
+import { ChangeEvent, useState } from 'react';
+import PropertyInfo from '../../enums/PropertyInfo.enum';
+import { usePropertyContext } from '../../hooks/PropertyContext';
+import style from "./_index.module.scss";
+
+type Props = {
+  filterPropertyByType: PropertyInfo;
+}
+
+function Search(props: Props) {
+  const { filterPropertyByType } = props;
+  const { filterProperties, properties, refreshProperties } = usePropertyContext();
+  const { colorScheme } = useMantineColorScheme();
+  const theme = useMantineTheme();
+  const [ userInput, setUserInput ] = useState<string>();
+
+  function handleSearch() {
+    if (userInput && properties) {
+      filterProperties(properties, filterPropertyByType, userInput);
+    } else {
+      console.log("Please input a value");
+    }
+  }
+
+  function handleUserInput(e: ChangeEvent<HTMLInputElement>) {
+    if (e.currentTarget.value === '') {
+      refreshProperties();
+    } else {
+      setUserInput(e.currentTarget.value);
+    }
+  }
+   
+  return (
+    <TextInput
+      className={`${style.searchContainer} ${style[colorScheme]}`}
+      icon={<IconSearch size={18} stroke={1.5} />}
+      radius="xl"
+      size="md"
+      onChange={handleUserInput}
+      rightSection={
+        <ActionIcon 
+          size={32} 
+          radius="xl" 
+          color={theme.primaryColor} 
+          variant="filled" 
+          onClick={handleSearch}
+        >
+          {theme.dir === 'ltr' ? (
+            <IconArrowRight size={18} stroke={1.5} />
+          ) : (
+            <IconArrowLeft size={18} stroke={1.5} />
+          )}
+        </ActionIcon>
```

Comment:
Place as a separate element outside the return statement.

Reply:
Resolved by removing icon arrow left instead since it's not used anyway.